ocaml/xapi/xapi_vm_helpers.ml

```diff
@@ -361,6 +361,14 @@ let assert_enough_memory_available ~__context ~self ~host ~snapshot =
           Int64.to_string host_mem_available;
         ]))
 
+(* CA-233580: prevent starting a control domain on a host different from its affinity*)
+let assert_matches_control_domain_affinity ~__context ~self ~host =
+  if Db.VM.get_is_control_domain ~__context ~self then
+    match Db.VM.get_affinity ~__context ~self with
+     | x when x = Ref.null || x = host -> ()
```

Comment:
`ocp-indent` does not agree with this indentation. The use of `match` instead of `if` is slightly dubious but I'm fine with it.